### DISCUSSION TITLE
Running both a server and a client?

### DISCUSSION THREAD

**Discussion by Br3nnabee** [cBournhonesque/lightyear#750] [Q&A] [answered]
Hi, 

Sorry, this is going to be extremely dumb. 

A few months ago while looking at this library, I distinctly remember being able to run both client and server versions of the same examples, however now I get the (somewhat expected) issue of it trying to compile one or the other and not at the same time.

Am I missing something? Did it never work that way in the past? If it did, what might've changed? 

Thanks in advance 🤦

**Comment by cBournhonesque** [comment, marked as answer]
Hi!
You can still run all examples with both the Server and Client enabled; see instructions here: https://github.com/cBournhonesque/lightyear/tree/main/examples#running-an-example

You can either run:
- in HostServer mode: the ClientPlugin and ServerPlugin are added to the same World
- in separate mode: two bevy Worlds are created, one with ClientPlugin and the other with ServerPlugin (and no gui)

Yes the commands to run the examples have changed recently. It's probably why you were confused :)

**Reply by Br3nnabee** [reply]
Thanks for the quick reply, I hope you've had a wonderful new years!

Does this new format not work with the simple setup example? I'm getting the below error, is this something on my end or worth making an issue for?

`thread 'main' panicked at examples\simple_setup\src\server.rs:54:13:
Error adding plugin bevy_core::TaskPoolPlugin in group bevy_internal::default_plugins::MinimalPlugins: plugin was already added in application`

**Reply by Br3nnabee** [reply]
Scratch that, I guess it's to do with the common. Did confuse me for a little, but thanks!

**Reply by cBournhonesque** [reply]
Correct, the simple_setup example is different and doesn't use `examples::common`.
The `examples::common` setup can be quite complex/confusing so the simple_set just shows a very simple way of creating the ClientPlugin and ServerPlugin
